src/views/Setup.vue

```diff
@@ -1,20 +1,98 @@
 <template>
   <div class="container">
-    <Card class="setup">
-      <div>
-        <p>We are preparing Sheikah</p>
+    <div class="center">
+      <div class="header">
+        <div class="filling-icon">
+          <font-awesome-icon
+            v-if="
+              setupMessage === 'Downloading wallet' ||
+                setupMessage === 'wallet up to date'
+            "
+            class="icon"
+            icon="wallet"
+          />
+          <font-awesome-icon
+            v-if="setupMessage === 'Running wallet'"
+            class="icon"
+            icon="cogs"
+          />
+          <font-awesome-icon
+            v-if="setupMessage === 'loaded'"
+            class="icon"
+            icon="check"
+          />
+          <div class="banner" />
+        </div>
+        <div>
+          <p class="progress-title">We are preparing Sheikah</p>
+          <p class="progress-subtitle">
+            {{ format(percentage) }}
+            <DotsLoading
+              v-if="!synced"
+              color="#444258"
+              data-test="loading-spinner"
+              class="spinner"
+            />
+          </p>
+        </div>
       </div>
-    </Card>
+      <el-progress
+        :text-inside="true"
+        :stroke-width="20"
+        :percentage="percentage"
+      />
+    </div>
   </div>
 </template>
 
 <script>
-import Card from '@/components/card/Card.vue'
+import { mapState, mapMutations } from 'vuex'
+import DotsLoading from '@/components/DotsLoading.vue'
 
 export default {
   name: 'WalletNotFound',
-  components: { Card },
-  methods: {},
+  components: {
+    DotsLoading,
+  },
+  computed: {
+    ...mapState({
+      setupMessage: state => state.uiInteractions.setupMessage,
+      client: state => state.wallet.api.client,
+      progress: state => state.uiInteractions.setupProgress,
+    }),
+    percentage() {
+      if (this.setupMessage === 'Downloading wallet') {
+        return this.progress / 4
+      } else if (this.setupMessage === 'wallet up to date') {
+        return 60
+      } else if (this.setupMessage === 'Running wallet') {
+        return 66
+      } else if (this.setupMessage === 'loaded') {
+        return 100
+      } else {
+        return 0
+      }
+    },
+  },
+  beforeDestroy() {
+    this.cleanMessage()
+  },
+  methods: {
+    ...mapMutations({
+      cleanMessage: 'cleanMessage',
+    }),
+    format(percentage) {
+      if (percentage <= 33) {
+        return 'Downloading wallet'
+      } else if (percentage <= 60 && percentage >= 33) {
+        return 'The wallet is up to date'
+      } else if (percentage <= 70) {
+        return 'Running wallet'
+      } else if (percentage > 70) {
+        return 'We are preparing Sheikah'
+      }
+    },
+  },
```

Comment:
What do you think about this?

```suggestion
    format(percentage) {
      if (percentage <= 80) {
        return 'Downloading wallet'
      } else if (percentage <= 90) {
        return 'Running wallet'
      } else if (percentage > 90) {
        return 'We are preparing Sheikah'
      }
    },
  },
```